framework/src/service/ServiceHooks.cpp

```diff
@@ -75,10 +74,8 @@ void ServiceHooks::RemovedService(const ServiceReference<ServiceListenerHook>& /
 
 void ServiceHooks::Open()
 {
-  {
-    auto l = this->Lock(); US_UNUSED(l);
-    listenerHookTracker.reset(new ServiceTracker<ServiceListenerHook>(GetBundleContext(), this));
-  }
+  auto l = this->Lock(); US_UNUSED(l);
+  listenerHookTracker.reset(new ServiceTracker<ServiceListenerHook>(GetBundleContext(), this));
```

Comment:
That is a good question - I am not sure yet. I was trying to see how we could end up in such a race condition in the first place but I need more time for this. Also, the lock scope originally enclosed the full function body but was then changed. Maybe it was due to the callback issue, but I do not remember. Will investigate this further...